src/test/java/io/confluent/connect/elasticsearch/ValidatorTest.java

```diff
@@ -313,12 +323,29 @@ public void testValidSsl() {
     props.put(SSL_CONFIG_PREFIX + SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, "b");
     props.put(SSL_CONFIG_PREFIX + SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, "c");
     props.put(SSL_CONFIG_PREFIX + SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, "d");
-    validator = new Validator(props);
+    validator = new Validator(props, mockClient);
 
     result = validator.validate();
     assertNoErrors(result);
   }
 
+  @Test
+  public void testValidConnection() {
+    validator = new Validator(props, mockClient);
+
+    Config result = validator.validate();
+    assertNoErrors(result);
+  }
+
+  @Test
+  public void testInvalidConnection() throws IOException {
```

Comment:
Add another test, that asserts the validator's behavior when `client.ping()` throws an exception?